### PR TITLE
Xcode 13 support, fix some warnings, remove suppressions

### PR DESCRIPTION
**Pull request by dflems**

### CODE REVIEW

File: Sources/SPTPersistentCache.m [hunk 1]
```diff
@@ -61,6 +61,13 @@ void SPTPersistentCacheSafeDispatch(_Nullable dispatch_queue_t queue, _Nonnull d
     }
 }
 
+@interface SPTPersistentCacheFileInfo : NSObject
+@property (nonatomic, strong, readonly) NSString *fileName;
+@property (nonatomic, strong, readonly) NSDate *mdate;
+@property (nonatomic, assign, readonly) off_t fileSize;
+- (instancetype)initWithFileName:(NSString *)fileName mdate:(NSDate *)mdate fileSize:(off_t)fileSize;
+@end
```
Comment (dflems) [inline Sources/SPTPersistentCache.m:69]: previously using a messy dictionary structure for what amounted to 3 properties (likely no worse performance-wise than nested dictionaries)

File: ci/spotify_os.xcconfig [hunk 2]
```diff
@@ -18,10 +18,10 @@
...
 // A set of configurations used for Spotify Objective-C Open Source Projects
 
-IPHONEOS_DEPLOYMENT_TARGET = 9.0
-WATCHOS_DEPLOYMENT_TARGET = 2.0
-TVOS_DEPLOYMENT_TARGET = 9.0
-MACOSX_DEPLOYMENT_TARGET = 10.10
+IPHONEOS_DEPLOYMENT_TARGET = 12.0
+WATCHOS_DEPLOYMENT_TARGET = 3.0
+TVOS_DEPLOYMENT_TARGET = 12.0
+MACOSX_DEPLOYMENT_TARGET = 10.12
```
Comment (dflems) [inline ci/spotify_os.xcconfig:24]: update deployment targets

File: ci/spotify_os.xcconfig [hunk 3]
```diff
@@ -75,11 +75,7 @@ YES_FOR_RELEASE = $(YES_FOR_RELEASE_$(CONFIGURATION))
 // Warnings:
 // Attribution: https://github.com/jonreid/XcodeWarnings
 
-WARNING_CFLAGS = -Weverything -Wno-error=deprecated -Wno-objc-missing-property-synthesis -Wno-gnu-conditional-omitted-operand -Wno-gnu -Wno-documentation-unknown-command -Wno-reserved-id-macro -Wno-auto-import -Wno-missing-variable-declarations -Wno-c++98-compat -Werror -Wno-direct-ivar-access -Wno-padded $(WARNING_CFLAGS_XCODE_$(XCODE_VERSION_MAJOR))
-
-WARNING_CFLAGS_XCODE_1000 = -Wno-objc-messaging-id
-WARNING_CFLAGS_XCODE_1100 = $(WARNING_CFLAGS_XCODE_1000)
-WARNING_CFLAGS_XCODE_1200 = $(WARNING_CFLAGS_XCODE_1000)
+WARNING_CFLAGS = -Weverything -Wno-error=deprecated -Wno-objc-missing-property-synthesis -Wno-gnu-conditional-omitted-operand -Wno-gnu -Wno-documentation-unknown-command -Wno-reserved-id-macro -Wno-auto-import -Wno-missing-variable-declarations -Wno-c++98-compat -Werror -Wno-direct-ivar-access -Wno-padded
```
Comment (dflems) [inline ci/spotify_os.xcconfig:78]: remove suppression on `-Wobjc-messaging-id`

**Review by rastersize — APPROVED** [review]

**Comment by adamhowardprice** [comment]
LGTM 👋

**Pull request merged**